exonum-light-client/src/test/java/com/exonum/client/ExonumHttpClientBlocksIntegrationTest.java

```diff
@@ -0,0 +1,364 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.client.response.Block;
+import com.exonum.client.response.BlockResponse;
+import com.exonum.client.response.BlocksResponse;
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ExonumHttpClientBlocksIntegrationTest {
+  private static MockWebServer server;
+  private static ExonumClient exonumClient;
+
+  @BeforeAll
+  static void start() throws IOException {
+    server = new MockWebServer();
+    server.start();
+
+    exonumClient = ExonumClient.newBuilder()
+        .setExonumHost(server.url("/").url())
+        .build();
+  }
+
+  @AfterAll
+  static void shutdown() throws IOException {
+    server.shutdown();
+  }
+
+  @Test
+  void getBlockByHeight() throws InterruptedException {
+    // Mock response
+    String previousHash = "fd510fc923683a4bb77af8278cd51676fbd0fcb25e2437bd69513d468b874bbb";
+    String txHash = "336a4acbe2ff0dd18989316f4bc8d17a4bfe79985424fe483c45e8ac92963d13";
+    String stateHash = "79a6f0fa233cc2d7d2e96855ec14bdcc4c0e0bb1a99ccaa912a555441e3b7512";
+    String tx1 = "336a4acbe2ff0dd18989316f4bc8d17a4bfe79985424fe483c45e8ac92963d13";
+    String time = "2019-02-14T14:12:52.037255Z";
+    String mockResponse = "{\n"
+        + "    \"block\": {\n"
+        + "        \"proposer_id\": 3,\n"
+        + "        \"height\": 1,\n"
+        + "        \"tx_count\": 1,\n"
+        + "        \"prev_hash\": \"" + previousHash + "\",\n"
+        + "        \"tx_hash\": \"" + txHash + "\",\n"
+        + "        \"state_hash\": \"" + stateHash + "\"\n"
+        + "    },\n"
+        + "    \"precommits\": [\"a410964c2c21199b48e2\"],\n"
+        + "    \"txs\": [\"" + tx1 + "\"],\n"
+        + "    \"time\": \"" + time + "\"\n"
+        + "}";
+    server.enqueue(new MockResponse().setBody(mockResponse));
+
+    // Call
+    long height = Long.MAX_VALUE;
+    BlockResponse response = exonumClient.getBlockByHeight(height);
+
+    // Assert response
+    assertThat(response.getBlock().getHeight(), is(1L));
+    assertThat(response.getBlock().getProposerId(), is(3));
```

Comment:
It's OK, but wouldn't this be clearer:
```
Block expectedBlock = aBlock()
  .height(1)
  .proposerId(3)
  //…
  .build();
assertThat(response.getBlock(), is(expectedBlock));
```

If a builder is too much work, I'd at least extract `response.getBlock()` in a variable.